src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -461,28 +459,52 @@ private EntityPlayerMP parseAnyPlayerValue(@NotNull final MinecraftServer server
     private IColony parseColonyValue(@NotNull final ICommandSender sender, @NotNull final MineColonyDataProvider mineColonyDataProvider, final String potentialArgumentValue)
     {
         final List<String> colonyNumberStrings = getColonyIdStrings(mineColonyDataProvider);
-        final Integer result = Ints.tryParse(potentialArgumentValue);
-        if (null != result)
-        {
-            int colonyNumber = result.intValue();
-            if (sender instanceof EntityPlayer && colonyNumber == -1)
-            {
-                final IColony icolony = mineColonyDataProvider.getIColonyByOwner(sender.getEntityWorld(), (EntityPlayer) sender);
-                if (icolony != null)
-                {
-                    colonyNumber = icolony.getID();
-                }
-            }
-            if (colonyNumberStrings.contains(String.valueOf(colonyNumber)))
-            {
-                return mineColonyDataProvider.getColony(colonyNumber, sender.getEntityWorld().provider.getDimension());
-            }
-            return null;
-        }
-        else
-        {
-            return null;
-        }
+        int dimensionId;
+        int colonyNumber;
+        if (potentialArgumentValue.contains("|"))
+        {
+        	String[] split = potentialArgumentValue.split(Pattern.quote("|"));
```

Comment:
Can't we give it '|' directly for split?

Reply:
Nope, it accepts regexes, and in that, | is the "or" character

Reply:
Ahh, good point